py/server/tests/test_liveness.py

```diff
@@ -80,6 +129,78 @@ def test_liveness_nested(self):
         self.assertFalse(nested_must_keep.j_table.tryRetainReference())
         self.assertFalse(nested_to_discard.j_table.tryRetainReference())
 
+    def test_liveness_siblings(self):
+        with liveness_scope() as l_scope:
+            to_discard = self.create_table()
+            df = to_pandas(to_discard)
+            must_keep = self.create_table()
+            df = to_pandas(must_keep)
+            l_scope.preserve(must_keep)
+
+        # Create a second scope and interact with it
+        with liveness_scope() as other_l_scope:
+            nested_to_discard = self.create_table()
+            df = to_pandas(nested_to_discard)
+            nested_must_keep = self.create_table()
+            df = to_pandas(nested_must_keep)
+            other_l_scope.preserve(nested_must_keep)
+
+        self.assertFalse(to_discard.j_table.tryRetainReference())
+        self.assertFalse(nested_to_discard.j_table.tryRetainReference())
+
+        self.assertTrue(nested_must_keep.j_table.tryRetainReference())
+        # drop extra reference
+        nested_must_keep.j_table.dropReference()
+
+        self.assertTrue(must_keep.j_table.tryRetainReference())
+        # drop extra reference
+        must_keep.j_table.dropReference()
+
+        self.assertFalse(to_discard.j_table.tryRetainReference())
+
+    def test_reopen_scope(self):
+        l_scope = LivenessScope()
+        with l_scope.open():
+            to_discard = self.create_table()
+            df = to_pandas(to_discard)
+            must_keep = self.create_table()
+            df = to_pandas(must_keep)
+            l_scope.preserve(must_keep)
+
+        self.assertTrue(to_discard.j_table.tryRetainReference())
+        # drop extra reference
+        to_discard.j_table.dropReference()
+
+        self.assertTrue(must_keep.j_table.tryRetainReference())
+        # drop extra reference
+        must_keep.j_table.dropReference()
+
+        # Reopen the scope and add to it again
+        with l_scope.open():
+            to_discard_2 = self.create_table()
+            df = to_pandas(to_discard_2)
+
+        self.assertTrue(to_discard.j_table.tryRetainReference())
+        # drop extra reference
+        to_discard.j_table.dropReference()
+
+        self.assertTrue(to_discard_2.j_table.tryRetainReference())
+        # drop extra reference
+        to_discard_2.j_table.dropReference()
+
+        self.assertTrue(must_keep.j_table.tryRetainReference())
+        # drop extra reference
+        must_keep.j_table.dropReference()
+
+        l_scope.release()
+
+        self.assertFalse(to_discard.j_table.tryRetainReference())
+        self.assertFalse(to_discard_2.j_table.tryRetainReference())
+
+        self.assertTrue(must_keep.j_table.tryRetainReference())
+        # drop extra reference
+        must_keep.j_table.dropReference()
+
```

Comment:
The `liveness_scope` pydoc indicates that `@liveness_scope` can be used to control scope, but I don't see a test for this case.

Reply:
added (and otherwise cleaned up tests a bit).

typehints were wrong for both contextmanager usages, i fixed them as well.